server/src/main/java/org/elasticsearch/search/rank/RankDoc.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.rank;
+
+import org.apache.lucene.search.ScoreDoc;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.VersionedNamedWriteable;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * {@code RankDoc} is the base class for all ranked results.
+ * Subclasses should extend this with additional information
+ * required for their global ranking method.
+ */
+public abstract class RankDoc extends ScoreDoc implements VersionedNamedWriteable {
```

Comment:
I think that this could implement Writeable, does not need to go further?

Reply:
Fixed.